armi/operators/settingsValidation.py

```diff
@@ -606,7 +606,7 @@ def decayCyclesHaveInputThatWillBeIgnored():
 
         self.addQuery(
             lambda: self.cs["geomFile"]
-            and self.coreSymmetry not in geometry.VALID_SYMMETRY,
+            and str(self.coreSymmetry) not in geometry.VALID_SYMMETRY,
             "{} is not a valid symmetry Please update symmetry on the geom file. "
```

Comment:
Something that ive been thinking would be nice is to implement some sort of symmetry/geom validator based on enums that is a bit more intelligent about which geoms support which symmetries without having to.... enumerate them. May be worth considering here.

Reply:
I guess there are two pieces to this. First, we have `geometry.VALID_SYMMETRY`, which is just a set of compatible pairs for what we now call `shape` (or `domain`) and `boundary`.  We could break up the list into pieces; i.e., for each `shape`, there is a list of valid `boundary`s, but that would probably be more complicated than the master list.

We also want to check the validity of `GeomType` with the `SymmetryType` (`full` works with anything, `hex` goes with `third periodic`, etc.) We currently don't have any check for this. I don't know how we would do it other than with another `set` like `VALID_SYMMETRY`, or a simple function that implements logic that defines the valid options.